src/core/version.js

```diff
@@ -1,85 +1,5 @@
 // @flow
 
-import deepFreeze from "deep-freeze";
-export type VersionInfo = {|
-  +major: number,
-  +minor: number,
-  +patch: number,
-  +gitState: GitState,
-  +environment: Environment,
-|};
-export type GitState = {|
-  +commitHash: string,
-  +commitTimestamp: string, // YYYYmmdd-HHMM, in commit-local time
-  +dirty: boolean, // does the worktree have unstaged/uncommitted changes?
-|};
-export type Environment = "development" | "production" | "test";
+const packageJSON = require("../../package.json");
```

Comment:
I’m curious—as you mentioned, this might pull `package.json` into the
frontend, which isn’t ideal (though not end of world); have you checked
whether it actually does, and if so whether destructuring here lets
Webpack tree-shake (`const {version: packageVersion} = require(…)`)?
I’m not familiar enough with Webpack to guess whether this will help.